OpenFeature.md

```diff
@@ -0,0 +1,103 @@
+Boolean variableValue = openFeatureClient.getBooleanValue(VARIABLE_KEY, false, context);
+```
+
+### Required Targeting Key
+
+For DevCycle SDK to work we require either a `targeting key` or `user_id` attribute to be set on the OpenFeature context.
+This value is used to identify the user as the `user_id` property for a `DevCycleUser` in DevCycle.
+
+### Mapping Context Properties to DevCycleUser
+
+The provider will automatically translate known `DevCycleUser` properties from the OpenFeature context to the `DevCycleUser` object.
+[DevCycleUser Java Interface](https://github.com/DevCycleHQ/java-server-sdk/blob/main/src/main/java/com/devcycle/sdk/server/common/model/DevCycleUser.java)
+
+For example all these properties will be set on the `DevCycleUser`:
+```java
+Map<String, Value> attributes = new LinkedHashMap<>();
+attributes.put("email", new Value("email@devcycle.com"));
+attributes.put("name", new Value("name"));
+attributes.put("country", new Value("CA"));
+attributes.put("language", new Value("en"));
+attributes.put("appVersion", new Value("1.0.11"));
+attributes.put("appBuild", new Value(1000));
+
+Map<String,Object> customData = new LinkedHashMap<>();
+customData.put("custom", "value");
+attributes.put("customData", new Value(Structure.mapToStructure(customData)));
+
+Map<String,Object> privateCustomData = new LinkedHashMap<>();
+privateCustomData.put("private", "data");
+attributes.put("privateCustomData", new Value(Structure.mapToStructure(privateCustomData)));
+
+EvaluationContext context = new ImmutableContext("test-1234", attributes);
+```
+
+Context properties that are not known `DevCycleUser` properties will be automatically
+added to the `customData` property of the `DevCycleUser`.
+
+### Context Limitations
+
+DevCycle only supports flat JSON Object properties used in the Context. Non-flat properties will be ignored.
+
+For example `obj` will be ignored:
+```java
+context = EvaluationContext(targeting_key="test-1234", attributes={
+    "obj": { "key": "value" }
+})
+```
+
+### JSON Flag Limitations
+
+The OpenFeature spec for JSON flags allows for any type of valid JSON value to be set as the flag value.
+
+For example the following are all valid default value types to use with OpenFeature:
+```java
+// Invalid JSON values for the DevCycle SDK, will return defaults
+openFeatureClient.getObjectValue("json-flag", new Value(new ArrayList<String>(Arrays.asList("value1", "value2"))));
+openFeatureClient.getObjectValue("json-flag", new Value(610));
+openFeatureClient.getObjectValue("json-flag", new Value(false));
+openFeatureClient.getObjectValue("json-flag", new Value("string"));
+```
+
+However, these are not valid types for the DevCycle SDK, the DevCycle SDK only supports JSON Objects:
+```java
+
+Map<String,Object> defaultJsonData = new LinkedHashMap<>();
+defaultJsonData.put("default", "value");
+openFeatureClient.getObjectValue("json-flag", new Value(Structure.mapToStructure(defaultJsonData)));
+```
```

Comment:
I did pretty much duplicate this in my devcycle-docs PR which is somewhat frustrating. It would be nice to have once source of truth but I do like having this stuff closer to the repo.  Maybe there is need for a hybrid solution where we make docs.devcycle.com the detailed version and we just do highlights in the SDK repos?